samila/functions.py

```diff
@@ -167,6 +298,45 @@ def save_data_file(data1, data2, matplotlib_version, file_adr):
     return result
 
 
+def save_config_file(g, matplotlib_version, file_adr):
+    """
+    Save config as file.
+
+    :param g: generative image instance
+    :type g: GenerativeImage
+    :param matplotlib_version: matplotlib version
+    :type matplotlib_version: str
+    :param file_adr: file address
+    :type file_adr: str
+    :return: result as dict
+    """
+    data = {}
+    if g.function1_str is not None:
```

Comment:
First of all, I think this line should be: `if g.function1_str is not None and g.function2_str is not None`.
And a question: If this condition is not met, should we continue to save the config file? I think we should warn users. Because in that case, the config file will not be useful!!

Reply:
Also `save_data_file` function

Reply:
Yes I think you're right.
I'll take care of that ASAP

Reply:
@sepandhaghighi I made a error scenario for this case.
As you mentioned it's meaningless if we save a config file without `function*_str`s or a data file without `data*`s.